### PR TITLE
Feature/project builds

### PR DESCRIPTION
**Pull request by torbrenner**
**What's in the PR**
* github actions for the builds of the different projects (currently samply, c4, dktk)
* new environment variables for the nNGM and DKTK
* removed hard coded urls from src/docker/samply_common_urls.xml
* Image is now based on newer tomcat image (9-jdk8-openjdk-slim) --> the image will only build for platform linux/amd64
* moved JMX Exporter and start.sh in docker container to /docker directory

**How to test manually**
* You can view the build actions on [my fork](https://github.com/torbrenner/share-client/actions?query=branch%3Afeature%2FprojectBuilds)
* The latest produced docker images are published at [Docker Hub](https://hub.docker.com/r/torbrenner/share-client) (With the merge github will push them to samply/share-client)

### CODE REVIEW

**Review by TasDeniz — CHANGES_REQUESTED** [review]
Please fix the merge conflict

**Review by TasDeniz — APPROVED** [review]

**Review by djuarezgf — CHANGES_REQUESTED** [review]

File: src/docker/start.sh [hunk 1]
```diff
@@ -2,40 +2,72 @@
...
+file=${CATALINA_HOME}/conf/${PROJECT}_common_config.xml
+sed -i "s~{proxy-url}~${HTTP_PROXY}~"                     "$file"
+sed -i "s/{proxy-user}/${PROXY_USER}/"                    "$file"
+sed -i "s/{proxy-pass}/${PROXY_PASS}/"                    "$file"
 
-sed -i "s#{store-url}#${STORE_URL}#"                      /usr/local/tomcat/conf/samply_common_urls.xml
-sed -i "s#{mdr-url}#${MDR_URL}#"                          /usr/local/tomcat/conf/samply_common_urls.xml
-sed -i "s#{directory-url}#${DIRECTORY_URL}#"              /usr/local/tomcat/conf/samply_common_urls.xml
+file=${CATALINA_HOME}/conf/${PROJECT}_common_urls.xml
+sed -i "s#{store-url}#${STORE_URL}#"                      "$file"
```
[the remark this answers is not in the document]
Comment (djuarezgf) [reply src/docker/start.sh:18]: @alexanderkiel : Was denkst Du?

[the remark this answers is not in the document]
Comment (torbrenner) [reply src/docker/start.sh:18]: I will create a seperate pull request for this.

File: src/docker/start.sh [hunk 2]
```diff
@@ -2,40 +2,72 @@
...
+sed -i "s/{postgres-host}/${POSTGRES_HOST}/"              "$file"
+sed -i "s/{postgres-port}/${POSTGRES_PORT:-5432}/"        "$file"
+sed -i "s/{postgres-db}/${POSTGRES_DB}/"                  "$file"
+sed -i "s/{postgres-user}/${POSTGRES_USER}/"              "$file"
+sed -i "s/{postgres-pass}/${POSTGRES_PASS}/"              "$file"
 
-sed -i "s~{proxy-url}~${HTTP_PROXY}~"                     /usr/local/tomcat/conf/samply_common_config.xml
-sed -i "s/{proxy-user}/${PROXY_USER}/"                    /usr/local/tomcat/conf/samply_common_config.xml
-sed -i "s/{proxy-pass}/${PROXY_PASS}/"                    /usr/local/tomcat/conf/samply_common_config.xml
+file=${CATALINA_HOME}/conf/${PROJECT}_common_config.xml
```
[the remark this answers is not in the document]
Comment (torbrenner) [reply src/docker/start.sh:12]: I will create a seperate pull request for this.

**Comment by djuarezgf** [comment]
I will allow this merge after second review at 17.2.2021

**Review by djuarezgf — APPROVED** [review]

**Pull request merged**